meta.go

```diff
@@ -83,6 +92,30 @@ func (m Meta) Check(log Logger, lvl Level, msg string) *CheckedMessage {
 // ErrorOutput. This method should only be used to report internal logger
 // problems and should not be used to report user-caused problems.
 func (m Meta) InternalError(cause string, err error) {
-	fmt.Fprintf(m.ErrorOutput, "%v %s error: %v\n", _timeNow().UTC(), cause, err)
+	fmt.Fprintf(m.ErrorOutput, "%v %s error: %v\n", time.Now().UTC(), cause, err)
 	m.ErrorOutput.Sync()
 }
+
+// Encode runs any Hook functions and then writes an encoded log entry to the
+// given io.Writer, returning any error.
+func (m Meta) Encode(w io.Writer, t time.Time, lvl Level, msg string, fields []Field) error {
+	enc := m.Encoder.Clone()
+	addFields(enc, fields)
```

Comment:
Not without conspiring with `Entry` like so:
```
commit 1d66d2a (HEAD -> imp_entry)
Author: Joshua T Corbin <joshua@uber.com>
Date:   5 minutes ago

    Have Meta and Entry conspire to golf away needless Encoder clones

diff --git a/entry.go b/entry.go
index 2f85457..e935850 100644
--- a/entry.go
+++ b/entry.go
@@ -33,9 +33,13 @@ type Entry struct {
 	Time    time.Time
 	Message string
 	enc     Encoder
+	myEnc   bool
 }
 
 // Fields returns a mutable reference to the entry's accumulated context.
-func (e Entry) Fields() KeyValue {
+func (e *Entry) Fields() KeyValue {
+	if !e.myEnc {
+		e.enc, e.myEnc = e.enc.Clone(), true
+	}
 	return e.enc
 }
diff --git a/meta.go b/meta.go
index 41c2e73..1787c3d 100644
--- a/meta.go
+++ b/meta.go
@@ -99,23 +99,30 @@ func (m Meta) InternalError(cause string, err error) {
 // Encode runs any Hook functions and then writes an encoded log entry to the
 // given io.Writer, returning any error.
 func (m Meta) Encode(w io.Writer, t time.Time, lvl Level, msg string, fields []Field) error {
-	enc := m.Encoder.Clone()
-	addFields(enc, fields)
+	enc, myEnc := m.Encoder, false
+	if len(fields) > 0 {
+		enc, myEnc = enc.Clone(), true
+		addFields(enc, fields)
+	}
 	if len(m.Hooks) >= 0 {
 		entry := _entryPool.Get().(*Entry)
 		entry.Level = lvl
 		entry.Message = msg
 		entry.Time = t
-		entry.enc = enc
+		entry.enc, entry.myEnc = enc, myEnc
 		for _, hook := range m.Hooks {
 			if err := hook(entry); err != nil {
 				m.InternalError("hook", err)
 			}
 		}
 		msg, enc = entry.Message, entry.enc
+		enc, myEnc = entry.enc, entry.myEnc
+		entry.enc, entry.myEnc = nil, false
 		_entryPool.Put(entry)
 	}
 	err := enc.WriteEntry(w, msg, lvl, t)
-	enc.Free()
+	if myEnc {
+		enc.Free()
+	}
 	return err
 }
```

I'm working on comparing benchmark results, so far adding this "improvement" slows some down, but I'm not sure what the noise level in the suite is yet...